dropshot/src/test_util.rs

```diff
@@ -39,8 +39,13 @@ use crate::server::{HttpServer, HttpServerStarter, ServerContext};
  * List of allowed HTTP headers in responses.  This is used to make sure we
  * don't leak headers unexpectedly.
  */
-const ALLOWED_HEADER_NAMES: [&str; 4] =
-    ["content-length", "content-type", "date", "x-request-id"];
+const ALLOWED_HEADER_NAMES: [&str; 5] = [
+    "content-length",
+    "content-type",
+    "date",
+    "x-request-id",
+    "transfer-encoding",
```

Comment:
Do we want to check for a particular value of the "transfer-encoding" header?

Context: These tests ensure that Dropshot returns correct responses, including the headers, not just the body.  The allowlist here tries to make sure we're not leaking unexpected headers.  It makes sense to add this header to the list.  If a header is in this list, presumably we also want to test that Dropshot is providing the correct value for it.

Reply:
Sure, I've added a test which checks for a specific value.

I'm not sure the test utility should care, admittedly. As an example, I believe "transfer-encoding = chunked" is an HTTP 1.1 feature, which HTTP 2.0 doesn't have - but most code acting on a response should work either way, since Hyper should be responsible for multiplexing the different implementations into the same stream-based API.

Reply:
I'm not sure I follow.  What I mean is: in general, "transfer-encoding" [appears to have only a handful of individual values](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Transfer-Encoding), plus comma-separated lists of those values.  If Dropshot sent "transfer-encoding: oxide", that would be broken.  It would also be broken if it sent "transfer-encoding: chunked" and then sent a response that wasn't chunked-encoded, although this is presumably done by hyper and such breakage would almost certainly cause the request to fail on the client anyway.

I could see asserting:

- that the value is either absent or "chunked" (knowing that we don't support the other values)
- that the value is absent unless we're hitting an API that's intentionally streaming.  Maybe this is what you're getting at -- it wouldn't be wrong if Dropshot chose to send non-streaming responses with "chunked", but it would be surprising and we might want to know if that happens without someone intentionally making that change.

Reply:
Sure, I'll add a toggle here for expecting headers with "any" value, or with a "known set of okay values". This will only permit "chunked" for now, but could let us permit a set of values in the future.